### ISSUE TITLE
About sklearn best estimator from cross validator

### ISSUE THREAD

**Issue by GermanCM** [rasbt/python-machine-learning-book-2nd-edition#114]
Dear professor Sebastian, 

Having a deeper look at the best estimator returned by a GridSearchCV instance from scikit learn, and also looking at the official documentation, I see that it says: 
"best_estimator_ : estimator or dict 
       Estimator that was chosen by the search, i.e. estimator which gave highest score"
so I wonder why it does not care also about the returned scores (let's say 10 if 10-fold) variance. I am thinking about a case where two model configurations give similar scores, but significant different variances (where the model with the second highest score has quite less variance in its predictions), shouldn't the best estimator be selected based on a weighted score/variance metric, rather than simply the highest score? It would mean a more reliable estimator.

Thanks in advance,
Germán

**Comment by rasbt** [comment]
That's a good point, and I agree with you -- unfortunately, this is currently  not implemented / supported in scikit-learn. Related to that, in practice, I also recommend the 1-standard error method, which basically means that you select the best model from k-fold based on pure performance, and then you select the simplest model that is within 1 standard error of that model. 

I have written about that in my lecture notes here: https://github.com/rasbt/stat479-machine-learning-fs19/blob/master/10_eval3-cv/10-eval3-cv__notes.pdf (section 10.10).
